javaslang/src/main/java/javaslang/Value.java

```diff
@@ -758,7 +758,7 @@ default CharSeq toCharSeq() {
      */
     default <K extends Comparable<? super K>, V> SortedMap<K, V> toSortedMap(Function<? super T, ? extends Tuple2<? extends K, ? extends V>> f) {
         Objects.requireNonNull(f, "f is null");
-        return toSortedMap((Comparator<? super K> & Serializable) K::compareTo, f);
+        return toSortedMap(Comparator.naturalOrder(), f);
```

Comment:
After 2.0.4 and 2.1.0 (maybe 2.2.0) are through the door I will focus on Java 9. These hints are very helpful!